idaes/models/properties/modular_properties/phase_equil/smooth_VLE_2.py

```diff
@@ -0,0 +1,278 @@
+# for full copyright and license information.
+#################################################################################
+"""
+Implementation of the formulation proposed in:
+
+Dabadghao, V., Ghouse, J., Eslick, J., Lee, A., Burgard, A., Miller, D., Biegler, L., 2022,
+A complementarity-based vapor-liquid equilibrium formulation for equation-oriented simulation
+and optimization. AIChE Journal, DOI: 10.1002/aic.18029
+"""
+
+from pyomo.environ import Constraint, Param, units as pyunits, Var, value
+
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.math import smooth_min
+from idaes.models.properties.modular_properties.base.utility import (
+    identify_VL_component_list,
+)
+from idaes.models.properties.modular_properties.base.utility import (
+    estimate_Tbub,
+    estimate_Tdew,
+)
+import idaes.core.util.scaling as iscale
+
+
+# -----------------------------------------------------------------------------
+class SmoothVLE2:
+    """
+    Improved Vapor-Liquid Equilibrium complementarity formulation for Cubic Equations of State.
+    """
+
+    @staticmethod
+    def phase_equil(b, phase_pair):
+        # This method is called via StateBlock.build, thus does not need clean-up
+        # try/except statements
+        suffix = "_" + phase_pair[0] + "_" + phase_pair[1]
+
+        # Smooth VLE assumes a liquid and a vapor phase, so validate this
+        (
+            l_phase,
+            v_phase,
+            vl_comps,
+            henry_comps,
+            l_only_comps,
+            v_only_comps,
+        ) = identify_VL_component_list(b, phase_pair)
+
+        if l_phase is None or v_phase is None:
+            raise ConfigurationError(
+                f"{b.params.name} Generic Property Package phase pair {phase_pair[0]}-{phase_pair[1]} "
+                "was set to use Smooth VLE formulation, however this is not a vapor-liquid pair."
+            )
+
+        # Definition of equilibrium temperature for smooth VLE
+        uom = b.params.get_metadata().default_units
+        t_units = uom.TEMPERATURE
+        f_units = uom.AMOUNT / uom.TIME
+
+        s = Var(
+            b.params.phase_list,
+            initialize=0.0,
+            bounds=(0, None),
+            doc="Slack variable for equilibrium temperature",
+            units=pyunits.dimensionless,
+        )
+        b.add_component("s" + suffix, s)
+
+        # Equilibrium temperature
+        def rule_teq(b):
+            if b.params.get_phase(phase_pair[0]).is_vapor_phase():
+                vapor_phase = phase_pair[0]
+                liquid_phase = phase_pair[1]
+            else:
+                vapor_phase = phase_pair[1]
+                liquid_phase = phase_pair[0]
+            return (
+                b._teq[phase_pair]
+                - b.temperature
+                - s[vapor_phase] * t_units
+                + s[liquid_phase] * t_units
+                == 0
```

Comment:
Why make the slack variable `s` dimensionless when we're multiplying it by `t_units` here? That also implies it should be scaled.